data/transactions/transaction.go

```diff
@@ -253,18 +261,48 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 		}
 
 	case protocol.KeyRegistrationTx:
+		if proto.EnableKeyregCoherencyCheck {
+			// ensure that the VoteLast is greater or equal to the VoteFirst
+			if tx.KeyregTxnFields.VoteFirst > tx.KeyregTxnFields.VoteLast {
+				return errKeyregTxnFirstVotingRoundGreaterThanLastVotingRound
+			}
+
+			// The trio of [VotePK, SelectionPK, VoteKeyDilution] needs to be all zeros or all non-zero for the transaction to be valid.
+			if !((tx.KeyregTxnFields.VotePK == crypto.OneTimeSignatureVerifier{} && tx.KeyregTxnFields.SelectionPK == crypto.VRFVerifier{} && tx.KeyregTxnFields.VoteKeyDilution == 0) ||
+				(tx.KeyregTxnFields.VotePK != crypto.OneTimeSignatureVerifier{} && tx.KeyregTxnFields.SelectionPK != crypto.VRFVerifier{} && tx.KeyregTxnFields.VoteKeyDilution != 0)) {
+				return errKeyregTxnNonCoherentVotingKeys
+			}
+
+			// if it's a going offline transaction
+			if tx.KeyregTxnFields.VoteKeyDilution == 0 {
+				// check that we don't have any VoteFirst/VoteLast fields.
+				if tx.KeyregTxnFields.VoteFirst != 0 || tx.KeyregTxnFields.VoteLast != 0 {
```

Comment:
(VoteFirst == 0 && VoteLast == 0) is incompatible with the above (VoteFirst > VoteLast)

Reply:
why ? for going offline transaction, `tx.KeyregTxnFields.VoteFirst > tx.KeyregTxnFields.VoteLast` is false for (0,0).
and we still want to check that both VoteFirst and VoteLast are explicitly zero.